src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1776,7 +1776,8 @@ protected internal virtual TypeDefinition MarkType (TypeReference reference, Dep
 			MarkSerializable (type);
 
 			// This marks static fields of KeyWords/OpCodes/Tasks subclasses of an EventSource type.
-			if (BCL.EventTracingForWindows.IsEventSourceImplementation (type, _context)) {
+			// The special handling of EventSource is still needed in .NET6 in library mode
+			if ((_context.GetTargetRuntimeVersion () < TargetRuntimeVersion.NET6 || !_context.IsOptimizationEnabled (CodeOptimizations.RemoveEventSourceSpecialHandling, type)) && BCL.EventTracingForWindows.IsEventSourceImplementation (type, _context)) {
```

Comment:
Could we avoid spreading these policy decisions throughout the code, and instead move the TFM check to where we decide which optimizations are enabled?

Reply:
I prefer to keep this here since we will need to treat `EventSource `as special for lower TFM as long as we support backward compat in the trimmer